homework/index.js

```diff
@@ -29,19 +29,144 @@
     });
     return elem;
   }
+  /* cSpell:enable */
+  // Creating the repository-details part
+  function createHyfRepoDetails(leftDiv, hyfRepo) {
+    const table = createAndAppend('table', leftDiv, { class: 'table' });
+    const tBody = createAndAppend('tbody', table);
+    const details = ['Repository', 'Description', 'Forks', 'Updated'];
+    details.forEach(detail => {
+      const tr = createAndAppend('tr', tBody, { class: `${detail.toLowerCase()}-row` });
+      createAndAppend('td', tr, {
+        text: `${detail}: `,
+        class: `label ${detail.toLowerCase()}-head`,
+      });
+      createAndAppend('td', tr, { class: `${detail.toLowerCase()}-data` });
+    });
+    const repoName = document.querySelector('.repository-data');
+    createAndAppend('a', repoName, {
+      href: hyfRepo.html_url,
+      target: '_blank',
+      text: hyfRepo.name,
+      class: 'repo-name',
+    });
+    if (hyfRepo.description === null) {
+      document.querySelector('.description-row').setAttribute('class', 'hide-description-row');
+    } else {
+      document.querySelector('.description-data').innerText = hyfRepo.description;
+    }
+    document.querySelector('.forks-data').innerText = hyfRepo.forks;
+    document.querySelector('.updated-data').innerText = new Date(
+      hyfRepo.updated_at,
+    ).toLocaleString();
+  }
+
+  // Creating the contributions part
+  function createContributors(rightDiv, url) {
+    fetchJSON(url, (err, contributors) => {
+      createAndAppend('h3', rightDiv, {
+        text: 'Contributions: ',
+        class: 'contributor-header',
+      });
+      if (err) {
+        createAndAppend('div', rightDiv, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      } else if (!(contributors && contributors.length)) {
+        createAndAppend('div', rightDiv, {
+          text: 'No contributions found.',
+          class: 'alert-error',
+        });
+      } else {
```

Comment:
Rather than using an `if-then-else` here, your code becomes simpler if you 'return early' for the error cases. If there is an error there is no more to be done.

```js
if (err) {
  createAndAppend('div', rightDiv, {
    text: err.message,
    class: 'alert-error',
  });
  return;
}
if (!(contributors && contributors.length)) {
  createAndAppend('div', rightDiv, {
    text: 'No contributions found.',
    class: 'alert-error',
  });
  return;
}

const ul = createAndAppend('ul', rightDiv, {
```

Your code needs less indentation, which makes it more readable.

Reply:
i am just use pretter to rearrange the lines